pygeoapi/process/manager/base.py

```diff
@@ -0,0 +1,303 @@
+# =================================================================
+#
+# Authors: Tom Kralidis <tomkralidis@gmail.com>
+#
+# Copyright (c) 2020 Tom Kralidis
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+from datetime import datetime
+import io
+import json
+import logging
+from multiprocessing import dummy
+import os
+
+from pygeoapi.util import DATETIME_FORMAT, JobStatus
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseManager:
+    """generic Manager ABC"""
+
+    def __init__(self, manager_def):
+        """
+        Initialize object
+
+        :param manager_def: manager definition
+
+        :returns: `pygeoapi.process.manager.base.BaseManager`
+        """
+
+        self.name = manager_def['name']
+        self.connection = manager_def['connection']
+        self.output_dir = manager_def.get('output_dir', None)
+
+    def create(self):
+        """
+        Create manager
+
+        :returns: `bool` status of result
+        """
+
+        raise NotImplementedError()
+
+    def connect(self):
```

Comment:
This is currently only used in the tinydb manager. Maybe leave it out of the base class?

Reply:
Updated.